libs/components/src/next/form/datepicker/vl-datepicker.component.ts

```diff
@@ -0,0 +1,338 @@
+            maxTime: { type: String, attribute: 'max-time' },
+            amPm: { type: String, attribute: 'am-pm' },
+            pattern: { type: String },
+        };
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+
+        // Fix tot <-> tot-en-met
+        if (Dutch?.nl) {
+            Dutch.nl.rangeSeparator = ' tot en met ';
+            flatpickr.l10ns.default.rangeSeparator = ' tot en met ';
+        }
+
+        if (!this.initialValue) {
+            this.initialValue = this.value;
+        }
+    }
+
+    protected firstUpdated(changedProperties: Map<string, unknown>): void {
+        super.firstUpdated(changedProperties);
+
+        this.initializeComponent();
+    }
+
+    updated(changedProperties: Map<string, unknown>): void {
+        super.updated(changedProperties);
+
+        const options = this.getDynamicOptions();
+        this.updateOptionsForInstance(options);
+
+        if (changedProperties.has('value')) {
+            this.updateValue(this.value);
+        }
+
+        if (changedProperties.has('block')) {
+            if (this.block) {
+                this.getFlatpickrWrapper()?.classList.add('flatpickr-wrapper--block');
+            } else {
+                this.getFlatpickrWrapper()?.classList.remove('flatpickr-wrapper--block');
+            }
+        }
+    }
+
+    disconnectedCallback() {
+        super.disconnectedCallback();
+
+        this.instance?.destroy();
+    }
+    protected render(): TemplateResult {
+        const inputClasses = {
+            'vl-input-field--error': this.error || this.isInvalid,
+            'vl-input-field--success': this.success,
+            'vl-input-field--block': this.block,
+            'vl-input-field--disabled': this.disabled,
+        };
+
+        const buttonClasses = {
+            'vl-input-addon--error': this.error || this.isInvalid,
+            'vl-input-addon--success': this.success,
+            'vl-input-addon--disabled': this.disabled,
+        };
+
+        return html`
+            <div class="vl-input-group" id="datepicker-wrapper">
+                <input
+                    id=${this.id}
+                    name=${this.name || this.id}
+                    type="text"
+                    class="vl-input-field js-vl-datepicker-input ${classMap(inputClasses)}"
+                    ?required=${this.required}
+                    ?disabled=${this.disabled}
+                    ?error=${this.error}
+                    ?success=${this.success}
+                    ?readonly=${this.readonly}
+                    pattern=${this.pattern}
+                    aria-label=${this.label}
+                    .value=${live(this.value)}
+                    @input=${this.onInput}
+                />
+                <button
+                    id="toggle-calendar"
+                    type="button"
+                    class="vl-input-addon js-vl-datepicker-toggle  ${classMap(buttonClasses)}"
```

Comment:
Kan je de `vl-input-addon js-vl-datepicker-toggle` classes mee in het buttonClasses object steken?

Reply:
Heb dit voor button & input zo gedaan, zodat die conditionele classes niet opnieuw geëvalueerd moeten worden bij elke render maar heb aangepast zoals voorgesteld gezien we hier tot nu toe geen onderscheid in maakten (en dit nooit iets zal uitmaken performance wise).